Tests/SwiftParserTest/Linkage.swift

```diff
@@ -0,0 +1,130 @@
+#if canImport(Darwin)
+import Darwin
+import XCTest
+
+final class LinkageTest: XCTestCase {
+  // Assert that SwiftSyntax and SwiftParser do not introduce more link-time
+  // dependencies than are strictly necessary. We want to minimize our link-time
+  // dependencies. If this set changes - in particular, if it grows - consult
+  // a SwiftSyntax maintainer to see if there's a way to avoid adding the
+  // dependency.
+  func testLinkage() throws {
+    for i in 0..<_dyld_image_count() {
+      let name = try XCTUnwrap(_dyld_get_image_name(i))
+      let path = String(cString: name)
+      guard path.hasSuffix("SwiftParserTest") else {
+        continue
+      }
+
+      var baseURL = URL(fileURLWithPath: path)
+      while !baseURL.pathComponents.isEmpty, baseURL.pathExtension != "xctest" {
+        baseURL = baseURL.deletingLastPathComponent()
+      }
+
+      if baseURL.pathComponents.isEmpty {
+        XCTFail("Unable to determine path to enclosing xctest bundle")
+        return
+      }
+
+      baseURL = baseURL.deletingLastPathComponent()
+      let swiftSyntaxURL = baseURL.appendingPathComponent("SwiftSyntax.o")
+      try assertLinkage(of: swiftSyntaxURL) { linkages in
+        XCTAssertEqual(linkages, [
+          .library("-lobjc"),
+          .library("-lswiftCompatibilityConcurrency"),
```

Comment:
Should we allow-list all `-lswiftCompatibility` prefixes? For example, there's a `-lswiftCompatibility56` that just got introduced and will kick in sometimes.